Explicit namespaces for esQuery and esKuery

lizozom opened the pull request:
## Summary

Part of https://github.com/elastic/kibana/issues/56881

This PR applies the proposed resolution for #52374, trying to balance code readability, API discoverability, DX and documentation generation using ApiExtractor on the `esQuery` and `esKuery` namespaces of `data` plugin.

The solution involves:

- Exporting types directly from index.ts, with a prefix (consistent with core)
- Exporting utility functions and constants using a "namespace" (A simple object).

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/kibana-app-arch (Team:AppArch)

lizozom commented:
@elasticmachine merge upstream

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/)
* Commit: fe08d7a1cb6c26cddfcde9e505f609199d449997

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/","number":"25445","commit":"fe08d7a1cb6c26cddfcde9e505f609199d449997"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/)
* Commit: 3dee29594d89128d4b2147603f2fce0a401218e4

### History
* :broken_heart: [Build #25445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/) failed fe08d7a1cb6c26cddfcde9e505f609199d449997

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/","number":"25445","commit":"fe08d7a1cb6c26cddfcde9e505f609199d449997"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/","number":"25449","commit":"3dee29594d89128d4b2147603f2fce0a401218e4"}]}
PIPELINE-->

lizozom commented:
@XavierM 

SIEM is not marked as code owners (and I don;t know your GITHUB handle), so you didn't get an automatic notification, but I wanted to ket you know about some changes in this PR:

 * I removed the copy of x-pack/legacy/plugins/siem/common/typed_json.ts you had, as its now exported from kibana_utils 
 * Some places in import from deep folders withing a folder (i.e. src/plugins/data/common/es_query) or from the common folder. Imports should be made from the top level of a plugin, and from public or server , so I fixed a couple of those as well.

lukeelmers commented on src/plugins/kibana_utils/common/typed_json.ts:
```diff
@@ -0,0 +1,27 @@
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface JsonArray extends Array<JsonValue> {}
```
Would it make more sense to put this in `packages/kbn-utility-types`?

I could go either way, but this does feel like a pretty basic TS interface that I could see folks reaching for a lot, and `kbn-utility-types` seems designed to address this use case more than `kibana_utils`.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/)
* Commit: 7b559252ebaf8d2d2560d6c542ba028dce7abc73

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/reporting/export_types/csv/server.CSV Execute Job calls getScopedSavedObjectsClient with request containing decrypted headers</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_reporting_export_types_csv_server/Kibana_Pipeline___x_pack_intake_agent___CSV_Execute_Job_calls_getScopedSavedObjectsClient_with_request_containing_decrypted_headers)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
TypeError: _server.fieldFormats.FieldFormatsRegistry is not a constructor
    at Object.fieldFormatServiceFactory (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.test.js:84:38)
    at Promise.all (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts:98:54)
    at executeJob (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts:97:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/reporting/export_types/csv/server.CSV Execute Job calls getScopedSavedObjectsClient with request containing getBasePath() returning server's basePath if the job doesn't have one</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_reporting_export_types_csv_server/Kibana_Pipeline___x_pack_intake_agent___CSV_Execute_Job_calls_getScopedSavedObjectsClient_with_request_containing_getBasePath___returning_server_s_basePath_if_the_job_doesn_t_have_one)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
TypeError: _server.fieldFormats.FieldFormatsRegistry is not a constructor
    at Object.fieldFormatServiceFactory (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.test.js:84:38)
    at Promise.all (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts:98:54)
    at executeJob (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts:97:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/reporting/export_types/csv/server.CSV Execute Job calls getScopedSavedObjectsClient with request containing getBasePath() returning job's basePath if the job has one</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_reporting_export_types_csv_server/Kibana_Pipeline___x_pack_intake_agent___CSV_Execute_Job_calls_getScopedSavedObjectsClient_with_request_containing_getBasePath___returning_job_s_basePath_if_the_job_has_one)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
TypeError: _server.fieldFormats.FieldFormatsRegistry is not a constructor
    at Object.fieldFormatServiceFactory (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.test.js:84:38)
    at Promise.all (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts:98:54)
    at executeJob (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/reporting/export_types/csv/server/execute_job.ts:97:7)
```
</details>

---
and 40 more failures, only showing the first 3.

### History
* :green_heart: [Build #25449](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/) succeeded 3dee29594d89128d4b2147603f2fce0a401218e4
* :broken_heart: [Build #25445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/) failed fe08d7a1cb6c26cddfcde9e505f609199d449997

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/","number":"25445","commit":"fe08d7a1cb6c26cddfcde9e505f609199d449997"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/","number":"25449","commit":"3dee29594d89128d4b2147603f2fce0a401218e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/","number":"25582","commit":"7b559252ebaf8d2d2560d6c542ba028dce7abc73"}]}
PIPELINE-->

lizozom commented on x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals/index.tsx:
```diff
@@ -11,8 +11,7 @@ import { connect } from 'react-redux';
 import { Dispatch } from 'redux';
 import { ActionCreator } from 'typescript-fsa';
 
-import { esFilters, esQuery } from '../../../../../../../../../src/plugins/data/common/es_query';
```
@XavierM FYI

lizozom commented on x-pack/legacy/plugins/graph/public/types/workspace_state.ts:
```diff
@@ -6,15 +6,7 @@
 import { WorkspaceField, AdvancedSettings } from './app_state';
-
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-interface JsonArray extends Array<JsonValue> {}
```
@flash1293 FYI

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25610/)
* Commit: de84b677c87c917c410cfdc913ff018a7b7ea10e

### History
* :broken_heart: [Build #25582](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/) failed 7b559252ebaf8d2d2560d6c542ba028dce7abc73
* :green_heart: [Build #25449](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/) succeeded 3dee29594d89128d4b2147603f2fce0a401218e4
* :broken_heart: [Build #25445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/) failed fe08d7a1cb6c26cddfcde9e505f609199d449997

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/","number":"25445","commit":"fe08d7a1cb6c26cddfcde9e505f609199d449997"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/","number":"25449","commit":"3dee29594d89128d4b2147603f2fce0a401218e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/","number":"25582","commit":"7b559252ebaf8d2d2560d6c542ba028dce7abc73"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25610/","number":"25610","commit":"de84b677c87c917c410cfdc913ff018a7b7ea10e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25638/)
* Commit: e6a9c0ce037126f7a9419ced2d1c12ed875bd898

### History
* :green_heart: [Build #25610](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25610/) succeeded de84b677c87c917c410cfdc913ff018a7b7ea10e
* :broken_heart: [Build #25582](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/) failed 7b559252ebaf8d2d2560d6c542ba028dce7abc73
* :green_heart: [Build #25449](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/) succeeded 3dee29594d89128d4b2147603f2fce0a401218e4
* :broken_heart: [Build #25445](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/) failed fe08d7a1cb6c26cddfcde9e505f609199d449997

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25445/","number":"25445","commit":"fe08d7a1cb6c26cddfcde9e505f609199d449997"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25449/","number":"25449","commit":"3dee29594d89128d4b2147603f2fce0a401218e4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25582/","number":"25582","commit":"7b559252ebaf8d2d2560d6c542ba028dce7abc73"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25610/","number":"25610","commit":"de84b677c87c917c410cfdc913ff018a7b7ea10e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25638/","number":"25638","commit":"e6a9c0ce037126f7a9419ced2d1c12ed875bd898"}]}
PIPELINE-->
